statusHandler/presenter/instanceInfoGetters.go

```diff
@@ -41,3 +52,83 @@ func (psh *PresenterStatusHandler) GetCountLeader() uint64 {
 }
+
+// GetNodeName will return node name
+func (psh *PresenterStatusHandler) GetNodeName() string {
+	nodeName := psh.getFromCacheAsString(core.MetricNodeDisplayName)
+	if nodeName == "" {
+		nodeName = "noname"
+	}
+
+	return nodeName
+}
+
+// GetTotalRewardsValue will return total value of rewards and how rewards was increased on every second
+// Rewards estimation will be equal with :
+// (numSignedBlocks - numProposedBlocks) * communityPercentage * Rewards +
+// numProposedBlocks * leaderPercentage * Rewards
+func (psh *PresenterStatusHandler) GetTotalRewardsValue() (string, string) {
+	rewardsValue := psh.getBigIntFromStringMetric(core.MetricRewardsValue)
+	communityPercentage := psh.getFloatFromStringMetric(core.MetricCommunityPercentage)
+	leaderPercentage := psh.getFloatFromStringMetric(core.MetricLeaderPercentage)
+	numSignedBlocks := psh.getFromCacheAsUint64(core.MetricCountConsensusAcceptedBlocks)
+	numProposedBlocks := psh.getFromCacheAsUint64(core.MetricCountAcceptedBlocks)
+
+	rewardsCommunityCoefficient := math.Abs(float64(int(numSignedBlocks-numProposedBlocks))) * communityPercentage
+	consensusSignedBlocksRewards := big.NewFloat(rewardsCommunityCoefficient)
+	consensusSignedBlocksRewards.Mul(consensusSignedBlocksRewards, big.NewFloat(0).SetInt(rewardsValue))
+
+	rewardsLeaderCoefficient := float64(numProposedBlocks) * leaderPercentage
+	leaderProposedBlocksRewards := big.NewFloat(rewardsLeaderCoefficient)
+	leaderProposedBlocksRewards.Mul(leaderProposedBlocksRewards, big.NewFloat(0).SetInt(rewardsValue))
+
+	totalRewardsBigFloat := big.NewFloat(0).Add(consensusSignedBlocksRewards, leaderProposedBlocksRewards)
+	totalRewards := new(big.Int)
+	totalRewardsBigFloat.Int(totalRewards)
+
+	difRewards := big.NewInt(0).Sub(totalRewards, psh.totalRewardsOld)
+	psh.totalRewardsOld.Set(totalRewards)
+	totalRewards.Sub(totalRewards, difRewards)
+
+	return totalRewards.Text(10), difRewards.Text(10)
+}
+
+// CalculateRewardsPerHour will return an approximation of how many test ERDs a validator will earn per hour
+// Rewards estimation per hour will be equals with :
+// (changeToBeInConsensus - changeToBeLeader) * roundsPerHour * hitRate * communityPercentage * Rewards +
+// changeToBeLeader * roundsPerHour * hitRate * leaderPercentage * Rewards
+func (psh *PresenterStatusHandler) CalculateRewardsPerHour() string {
+	consensusGroupSize := psh.getFromCacheAsUint64(core.MetricConsensusGroupSize)
+	numValidators := psh.getFromCacheAsUint64(core.MetricNumValidators)
+	totalBlocks := psh.GetProbableHighestNonce()
+	rounds := psh.GetCurrentRound()
+	roundDuration := psh.GetRoundTime()
+	secondsInAHour := uint64(3600)
+	rewardsValue := psh.getBigIntFromStringMetric(core.MetricRewardsValue)
+	communityPercentage := psh.getFloatFromStringMetric(core.MetricCommunityPercentage)
+	leaderPercentage := psh.getFloatFromStringMetric(core.MetricLeaderPercentage)
+
+	if consensusGroupSize == 0 || numValidators == 0 || totalBlocks == 0 ||
```

Comment:
This if is not very pretty. We can pull this checks into a validation function, and if it errors we can return "0"

Reply:
Done.